src/chainstate/burn/distribution.rs

```diff
@@ -0,0 +1,827 @@
+            }
+        }
+
+        return;
+    }
+
+    /// Move a list of user burns into their burn sample points.
+    fn apply_user_burns(burn_sample: &mut Vec<BurnSamplePoint<A, K>>, user_burns: Vec<UserBurnSupportOp<A, K>>) -> () {
+        // map each burn sample's (VRF public key, block hash 160) pair to its index.
+        // The pair will be unique, since the DB requires each VRF key to be unique.
+        // Hovever, the block hash 160 is not guaranteed to be unique -- two different
+        // leaders can burn for the same block.
+        let mut burn_index : BTreeMap<(String, Hash160), usize> = BTreeMap::new();
+        for i in 0..burn_sample.len() {
+            let block_header_hash_160 = Hash160::from_sha256(&burn_sample[i].candidate.block_header_hash.as_bytes());
+            let vrf_hex = ECVRF_public_key_to_hex(&burn_sample[i].key.public_key);
+            assert!(burn_index.get(&(vrf_hex.clone(), block_header_hash_160)).is_none());
+
+            burn_index.insert((vrf_hex, block_header_hash_160), i);
+        }
+
+        // match user burns to the leaders
+        for user_burn in user_burns {
+            let block_idx_opt = burn_index.get(&(ECVRF_public_key_to_hex(&user_burn.public_key), user_burn.block_header_hash_160));
+            match block_idx_opt {
+                Some(ref_block_idx) => {
+                    // user burn matches a (key, block) pair committed to in this block
+                    let idx = *ref_block_idx;
+                    burn_sample[idx].burns += (user_burn.burn_fee as u128);
+                    burn_sample[idx].user_burns.push(user_burn);
+                },
+                None => {
+                    info!("User burn {} ({},{}) of {} for key={}, block={} has no matching block commit",
+                          &user_burn.txid.to_hex(), user_burn.block_number, user_burn.vtxindex, user_burn.burn_fee,
+                          ECVRF_public_key_to_hex(&user_burn.public_key), &user_burn.block_header_hash_160.to_hex());
+                    continue;
+                }
+            };
+        }
+    }
+
+    /// Calculate the ranges between 0 and 2**256 - 1 over which each point in the burn sample
+    /// applies, so we can later select which block to use.
+    fn make_sortition_ranges(burn_sample: &mut Vec<BurnSamplePoint<A, K>>) -> () {
+        if burn_sample.len() == 0 {
+            // empty sample
+            return;
+        }
+        if burn_sample.len() == 1 {
+            // sample that covers the whole range 
+            burn_sample[0].range_start = Uint256::zero();
+            burn_sample[0].range_end = Uint256::max();
+            return;
+        }
+
+        // total burns for valid blocks?
+        // NOTE: this can't overflow -- there's no way we get that many (u64) burns
+        let total_burns_u128: u128 = burn_sample
```

Comment:
should this just be a call to `get_total_burns` ?

Reply:
Not sure I understand what you're asking?